worlds/faxanadu/__init__.py

```diff
@@ -0,0 +1,204 @@
+
+
+class FaxanaduLocation(Location):
+    game: str = "Faxanadu"
+
+
+class FaxanaduItem(Item):
+    game: str = "Faxanadu"
+
+
+class FaxanaduWeb(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Faxanadu randomizer connected to an Archipelago Multiworld",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Daivuk"]
+    )]
+    theme = "dirt"
+
+
+class FaxanaduWorld(World):
+    """
+    Faxanadu is an action role-playing platform video game developed by Hudson Soft for the Nintendo Entertainment System
+    """
+    options_dataclass = FaxanaduOptions
+    options: FaxanaduOptions
+    game = "Faxanadu"
+    web = FaxanaduWeb()
+
+    item_name_to_id = {item.name: item.id for item in Items.items}
+    item_name_to_item = {item.name: item for item in Items.items}
+    location_name_to_id = {loc.name: loc.id for loc in Locations.locations}
+
+
+    def __init__(self, world: MultiWorld, player: int):
+        self.filler_ratios = {}
+        
+        super().__init__(world, player)
+
+
+    def create_regions(self):
+        Regions.create_regions(self)
+
+        # Add locations into regions
+        for region in self.multiworld.get_regions(self.player):
+            for loc in [location for location in Locations.locations if location.region == region.name]:
+                location = FaxanaduLocation(self.player, loc.name, loc.id, region)
+
+                # In Faxanadu, Poison hurts you when picked up. It makes no sense to sell them in shops
+                if loc.type == Locations.LocationType.shop:
+                    location.item_rule = lambda item, player=self.player: not (player == item.player and item.name == "Poison")
+                
+                region.locations.append(location)
+
+
+    def set_rules(self):
+        Rules.set_rules(self)
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Killed Evil One", self.player)
+    
+
+    def create_item(self, name: str) -> FaxanaduItem:
+        item: Items.ItemDef = self.item_name_to_item[name]
+        return FaxanaduItem(name, item.classification, item.id, self.player)
+    
+
+    # REturns how many red potions were prefilled into shops
+    def prefill_shop_red_potions(self) -> int:
+        red_potion_in_shop_count = 0
+        if self.options.keep_shop_red_potions.value:
+            red_potion_item = next(item for item in Items.items if item.name == "Red Potion")
+            red_potion_shop_locations = [
+                loc
+                for loc in Locations.locations
+                if loc.type == Locations.LocationType.shop and loc.original_item == Locations.ItemType.red_potion
+            ]
+            for loc in red_potion_shop_locations:
+                location = self.multiworld.get_location(loc.name, self.player)
+                if loc.type == Locations.LocationType.shop and loc.original_item == Locations.ItemType.red_potion:
```

Comment:
A mistake: Because original code didn't have the definition list (Which I hate), but a previous commenter asked me for it. And forgot that there after :)